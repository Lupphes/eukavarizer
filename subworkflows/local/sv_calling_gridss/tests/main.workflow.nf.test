nextflow_workflow {

    name "Test Workflow SV_CALLING_GRIDSS"
    script "../main.nf"
    workflow "SV_CALLING_GRIDSS"
    config "subworkflows/local/sv_calling_gridss/tests/nextflow.config"
    tag "subworkflows"
    tag "subworkflows_local"
    tag "sv_calling_gridss"

    /*
     * Test scenario 1: Illumina BAM input without repeat annotation
     * This test validates GRIDSS's basic SV calling functionality without the RepeatMasker annotation step.
     * Expected behavior: GRIDSS should process the BAM and generate VCF output with detected SVs,
     * along with compressed VCF (vcfgz), tabix index (tbi), and CSI index (csi) files.
     */
    test("Illumina BAM input - no annotation - stub") {
        options '-stub'

        when {
            params {
                gridss_annotate = false
                minimap2_flag = false
                long_read_threshold = 1000
            }
            workflow {
                """
                input[0] = channel.of([
                    [id: 'test', single_end: false, platform: 'illumina', median_bp: 150, sample: 'sample1', patient: 'patient1'],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true)
                ])
                input[1] = channel.value([[id: 'genome'], file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)])
                input[2] = channel.value([[id: 'genome'], file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta.fai', checkIfExists: true)])
                input[3] = channel.value([[id: 'genome'], []])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert workflow.out.vcf != null },
                { assert workflow.out.vcfgz != null },
                { assert workflow.out.tbi != null },
                { assert workflow.out.csi != null },
                { assert snapshot(workflow.out).match() }
            )
        }
    }

    /*
     * Test scenario 2: Illumina BAM input with RepeatMasker annotation
     * This test validates GRIDSS with the optional repeat masking annotation enabled.
     * Expected behavior: GRIDSS should call SVs and then annotate them with repeat masker data,
     * which helps identify variants located in repetitive genomic regions. This adds additional
     * annotation fields to the VCF output.
     */
    test("Illumina BAM input - with annotation - stub") {
        options '-stub'

        when {
            params {
                gridss_annotate = true
                minimap2_flag = false
                long_read_threshold = 1000
            }
            workflow {
                """
                input[0] = channel.of([
                    [id: 'test', single_end: false, platform: 'illumina', median_bp: 150, sample: 'sample1', patient: 'patient1'],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true)
                ])
                input[1] = channel.value([[id: 'genome'], file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)])
                input[2] = channel.value([[id: 'genome'], file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta.fai', checkIfExists: true)])
                input[3] = channel.value([[id: 'genome'], []])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert workflow.out.vcf != null },
                { assert workflow.out.vcfgz != null },
                { assert workflow.out.tbi != null },
            { assert workflow.out.csi != null }
            )
        }
    }

    /*
     * Test scenario 3: Multiple samples processed in batch
     * This test validates GRIDSS's ability to process multiple samples in parallel without annotation.
     * Expected behavior: GRIDSS should successfully call SVs for each sample independently,
     * generating separate VCF outputs for each sample (sample1 and sample2).
     */
    test("Multiple samples - stub") {
        options '-stub'

        when {
            params {
                gridss_annotate = false
                minimap2_flag = false
                long_read_threshold = 1000
            }
            workflow {
                """
                // Two Illumina paired-end samples from the same patient
                input[0] = channel.of(
                    [
                        [id: 'sample1', single_end: false, platform: 'illumina', median_bp: 150, sample: 'sample1', patient: 'patient1'],
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true)
                    ],
                    [
                        [id: 'sample2', single_end: false, platform: 'illumina', median_bp: 150, sample: 'sample2', patient: 'patient1'],
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true)
                    ]
                )
                input[1] = channel.value([[id: 'genome'], file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)])
                input[2] = channel.value([[id: 'genome'], file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta.fai', checkIfExists: true)])
                input[3] = channel.value([[id: 'genome'], []])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert workflow.out.vcf != null },
                { assert workflow.out.vcfgz != null },
                { assert workflow.out.tbi != null },
            { assert workflow.out.csi != null }
            )
        }
    }
}
