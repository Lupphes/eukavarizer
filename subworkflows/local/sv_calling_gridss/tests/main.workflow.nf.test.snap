{
    "Illumina BAM input - no annotation - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test-gridss-svync",
                            "single_end": false,
                            "platform": "illumina",
                            "median_bp": 150,
                            "sample": "sample1",
                            "patient": "patient1"
                        },
                        "re-sample1-illumina-false-gridss.vcf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test-gridss-svync",
                            "single_end": false,
                            "platform": "illumina",
                            "median_bp": 150,
                            "sample": "sample1",
                            "patient": "patient1"
                        },
                        "re-sample1-illumina-false-gridss.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test-gridss-svync",
                            "single_end": false,
                            "platform": "illumina",
                            "median_bp": 150,
                            "sample": "sample1",
                            "patient": "patient1"
                        },
                        "re-sample1-illumina-false-gridss.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test-gridss-svync",
                            "single_end": false,
                            "platform": "illumina",
                            "median_bp": 150,
                            "sample": "sample1",
                            "patient": "patient1"
                        },
                        "re-sample1-illumina-false-gridss.vcf.gz.csi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    "versions.yml:md5,1236b5bb110b075b93374c6872bdacaf",
                    "versions.yml:md5,3c231d0a0ed544893133b04f4287f871",
                    "versions.yml:md5,df40cdeae190bc8287bd86dc507dee9a"
                ],
                "csi": [
                    [
                        {
                            "id": "test-gridss-svync",
                            "single_end": false,
                            "platform": "illumina",
                            "median_bp": 150,
                            "sample": "sample1",
                            "patient": "patient1"
                        },
                        "re-sample1-illumina-false-gridss.vcf.gz.csi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "tbi": [
                    [
                        {
                            "id": "test-gridss-svync",
                            "single_end": false,
                            "platform": "illumina",
                            "median_bp": 150,
                            "sample": "sample1",
                            "patient": "patient1"
                        },
                        "re-sample1-illumina-false-gridss.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "vcf": [
                    [
                        {
                            "id": "test-gridss-svync",
                            "single_end": false,
                            "platform": "illumina",
                            "median_bp": 150,
                            "sample": "sample1",
                            "patient": "patient1"
                        },
                        "re-sample1-illumina-false-gridss.vcf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "vcfgz": [
                    [
                        {
                            "id": "test-gridss-svync",
                            "single_end": false,
                            "platform": "illumina",
                            "median_bp": 150,
                            "sample": "sample1",
                            "patient": "patient1"
                        },
                        "re-sample1-illumina-false-gridss.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1236b5bb110b075b93374c6872bdacaf",
                    "versions.yml:md5,3c231d0a0ed544893133b04f4287f871",
                    "versions.yml:md5,df40cdeae190bc8287bd86dc507dee9a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.8"
        },
        "timestamp": "2025-10-23T02:37:59.929035733"
    }
}